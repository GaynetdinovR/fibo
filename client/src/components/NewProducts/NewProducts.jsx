import styles from "../../styles/components/NewProducts.module.sass";
import H4 from "../../ui/H4.jsx";
import H5 from "../../ui/H5.jsx";
import { getRandom4NewProducts } from "../../scripts/functions.js";

const NewProducts = ({ newProducts }) => {
	const typeName = {
		salad: "Салат",
		pizza: "Пицца",
		pasta: "Паста",
		snack: "Закуска",
		soup: "Суп"
	};

	return (
		<aside className={styles.new_products}>
			<H4 className={styles.new_products__title}>Новинки</H4>
			<div className={styles.new_products__content}>
				{newProducts.map((product, i) => (
					<button
						className={styles.new_products__new_product}
						key={i}
					>
						<div className={styles.new_product__img}>
							<img src={product?.img_url} alt="product" />
						</div>
						<div className={styles.new_product__info}>
							<H5 className={styles.new_product__name}>
								{typeName[product?.type]}
							</H5>
							<span className={styles.new_product__price}>
								{product?.price} ₽
							</span>
						</div>
					</button>
				))}
			</div>
		</aside>
	);
};

export default NewProducts;
