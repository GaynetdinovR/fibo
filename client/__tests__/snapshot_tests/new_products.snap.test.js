import renderer from "react-test-renderer";
import { it, expect } from "@jest/globals";
import NewProducts from "../../src/components/NewProducts/NewProducts.jsx";

const products = [
	{
		name: "test",
		price: 299.99,
		img_url: "./content/pizzas/pizza_1.png",
		type: "pizza"
	},
	{
		name: "test",
		price: 455.99,
		img_url: "./content/pastas/pasta_1.png",
		type: "pasta"
	},
	{
		name: "test",
		price: 322.99,
		img_url: "./content/soups/soup_1.png",
		type: "soup"
	},
	{
		name: "test",
		price: 553.99,
		img_url: "./content/snacks/snack_1.png",
		type: "snack"
	}
];

it("renders correctly", () => {
	const tree = renderer.create(<NewProducts newProducts={products} />).toJSON();
	expect(tree).toMatchSnapshot();
});
