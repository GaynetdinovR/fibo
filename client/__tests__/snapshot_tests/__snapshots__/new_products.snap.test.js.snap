// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<aside
  className="new_products"
>
  <h4
    className="new_products__title h4"
  >
    Новинки
  </h4>
  <div
    className="new_products__content"
  >
    <button
      className="new_products__new_product"
    >
      <div
        className="new_product__img"
      >
        <img
          alt="product"
          src="./content/pizzas/pizza_1.png"
        />
      </div>
      <div
        className="new_product__info"
      >
        <h5
          className="new_product__name h5"
        >
          Пицца
        </h5>
        <span
          className="new_product__price"
        >
          299.99
           ₽
        </span>
      </div>
    </button>
    <button
      className="new_products__new_product"
    >
      <div
        className="new_product__img"
      >
        <img
          alt="product"
          src="./content/pastas/pasta_1.png"
        />
      </div>
      <div
        className="new_product__info"
      >
        <h5
          className="new_product__name h5"
        >
          Паста
        </h5>
        <span
          className="new_product__price"
        >
          455.99
           ₽
        </span>
      </div>
    </button>
    <button
      className="new_products__new_product"
    >
      <div
        className="new_product__img"
      >
        <img
          alt="product"
          src="./content/soups/soup_1.png"
        />
      </div>
      <div
        className="new_product__info"
      >
        <h5
          className="new_product__name h5"
        >
          Суп
        </h5>
        <span
          className="new_product__price"
        >
          322.99
           ₽
        </span>
      </div>
    </button>
    <button
      className="new_products__new_product"
    >
      <div
        className="new_product__img"
      >
        <img
          alt="product"
          src="./content/snacks/snack_1.png"
        />
      </div>
      <div
        className="new_product__info"
      >
        <h5
          className="new_product__name h5"
        >
          Закуска
        </h5>
        <span
          className="new_product__price"
        >
          553.99
           ₽
        </span>
      </div>
    </button>
  </div>
</aside>
`;
